shell/platform/darwin/macos/framework/Source/FlutterPlatformViewController.mm

```diff
@@ -0,0 +1,107 @@
+  NSMutableDictionary<NSString*, NSObject<FlutterPlatformViewFactory>*>* _platformViewFactories;
+
+  // Map from platform view id to the underlying NSView.
+  std::map<int, NSView*> _platformViews;
+
+  // View ids that are going to be disposed on the next present call.
+  std::unordered_set<int64_t> _platformViewsToDispose;
+}
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    _platformViewFactories = [[NSMutableDictionary alloc] init];
+  }
+  return self;
+}
+
+- (void)onCreateWithViewID:(int64_t)viewId
+                  viewType:(nonnull NSString*)viewType
+                    result:(nonnull FlutterResult)result {
+  if (_platformViews.count(viewId) != 0) {
+    result([FlutterError errorWithCode:@"recreating_view"
+                               message:@"trying to create an already created view"
+                               details:[NSString stringWithFormat:@"view id: '%lld'", viewId]]);
+  }
+
+  NSObject<FlutterPlatformViewFactory>* factory = _platformViewFactories[viewType];
+  if (!factory) {
+    result([FlutterError
+        errorWithCode:@"unregistered_view_type"
+              message:@"trying to create a view with an unregistered type"
+              details:[NSString stringWithFormat:@"unregistered view type: '%@'", viewType]]);
+    return;
+  }
+
+  NSView* platform_view = [factory createWithFrame:CGRectZero viewIdentifier:viewId arguments:nil];
+  _platformViews[viewId] = platform_view;
+  result(nil);
+}
+
+- (void)onDisposeWithViewID:(int64_t)viewId result:(nonnull FlutterResult)result {
+  if (_platformViews.count(viewId) == 0) {
+    result([FlutterError errorWithCode:@"unknown_view"
+                               message:@"trying to dispose an unknown"
+                               details:[NSString stringWithFormat:@"view id: '%lld'", viewId]]);
+    return;
+  }
+
+  // The following disposePlatformViews call will dispose the views.
+  _platformViewsToDispose.insert(viewId);
+  result(nil);
+}
+
+- (void)registerViewFactory:(nonnull NSObject<FlutterPlatformViewFactory>*)factory
+                     withId:(nonnull NSString*)factoryId {
+  _platformViewFactories[factoryId] = factory;
+}
+
+- (nullable NSView*)platformViewWithID:(int64_t)viewId {
+  if (_platformViews.count(viewId)) {
+    return _platformViews[viewId];
+  } else {
+    return nil;
+  }
+}
+
+- (void)handleMethodCall:(nonnull FlutterMethodCall*)call result:(nonnull FlutterResult)result {
+  if ([[call method] isEqualToString:@"create"]) {
+    NSMutableDictionary<NSString*, id>* args = [call arguments];
+    int64_t viewId = [args[@"id"] longValue];
+    NSString* viewType = [NSString stringWithUTF8String:([args[@"viewType"] UTF8String])];
+    [self onCreateWithViewID:viewId viewType:viewType result:result];
+  } else if ([[call method] isEqualToString:@"dispose"]) {
+    NSNumber* arg = [call arguments];
+    int64_t viewId = [arg longLongValue];
```

Comment:
Can we just use `longValue]`? Why longLongValue.

Reply:
Done!